README.rst
Python Client for Google Cloud Text-to-Speech API
=================================================

|stable| |pypi| |versions|

`Google Cloud Text-to-Speech API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-texttospeech.svg
   :target: https://pypi.org/project/google-cloud-texttospeech/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-texttospeech.svg
   :target: https://pypi.org/project/google-cloud-texttospeech/
.. _Google Cloud Text-to-Speech API: https://cloud.google.com/text-to-speech
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/texttospeech/latest
.. _Product Documentation:  https://cloud.google.com/text-to-speech

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the Google Cloud Text-to-Speech API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the Google Cloud Text-to-Speech API.:  https://cloud.google.com/text-to-speech
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-texttospeech


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-texttospeech

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for Google Cloud Text-to-Speech API
   to see other available methods on the client.
-  Read the `Google Cloud Text-to-Speech API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _Google Cloud Text-to-Speech API Product documentation:  https://cloud.google.com/text-to-speech
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

[PyPI History][1]

[1]: https://pypi.org/project/google-cloud-texttospeech/#history

## [2.15.0](https://github.com/googleapis/python-texttospeech/compare/v2.14.1...v2.15.0) (2023-09-13)


### Features

* Enable "rest" transport in Python for services supporting numeric enums ([#362](https://github.com/googleapis/python-texttospeech/issues/362)) ([52bcda6](https://github.com/googleapis/python-texttospeech/commit/52bcda6cd119a676bc182e900f8cb0e0c52e99b2))


### Bug Fixes

* Add async context manager return types ([#375](https://github.com/googleapis/python-texttospeech/issues/375)) ([8de4b09](https://github.com/googleapis/python-texttospeech/commit/8de4b09dbafd91e251b78981e838ddcfca989ff5))


### Documentation

* Fix formatting of request arg in docstring ([#367](https://github.com/googleapis/python-texttospeech/issues/367)) ([4747441](https://github.com/googleapis/python-texttospeech/commit/474744180b2a148c513ece4ba15fa66206c70d8d))
* Minor formatting ([b85a47c](https://github.com/googleapis/python-texttospeech/commit/b85a47c63e12810d7badbd3336a12db9eace917f))
* Update documentation to require certain fields ([#368](https://github.com/googleapis/python-texttospeech/issues/368)) ([7f4c79f](https://github.com/googleapis/python-texttospeech/commit/7f4c79f5633cc2b2d7425659c4a7e2067469fa88))

## [2.14.1](https://github.com/googleapis/python-texttospeech/compare/v2.14.0...v2.14.1) (2023-01-20)


### Bug Fixes

* Add context manager return types ([6415162](https://github.com/googleapis/python-texttospeech/commit/641516224405ac91e08d5fee161a5fcde4ed0ba0))


### Documentation

* Add documentation for enums ([6415162](https://github.com/googleapis/python-texttospeech/commit/641516224405ac91e08d5fee161a5fcde4ed0ba0))

## [2.14.0](https://github.com/googleapis/python-texttospeech/compare/v2.13.0...v2.14.0) (2023-01-10)


### Features

* Add support for python 3.11 ([#353](https://github.com/googleapis/python-texttospeech/issues/353)) ([91e5500](https://github.com/googleapis/python-texttospeech/commit/91e5500cde7939e5c8f37e3134d21742c741892a))

## [2.13.0](https://github.com/googleapis/python-texttospeech/compare/v2.12.3...v2.13.0) (2022-12-15)


### Features

* Add LRS API ([07227a0](https://github.com/googleapis/python-texttospeech/commit/07227a06b555d590fc35055ec229f22c6d8dd139))
* Add support for `google.cloud.texttospeech.__version__` ([a2a534c](https://github.com/googleapis/python-texttospeech/commit/a2a534ceccbad9114cb30599129eb168ab60cb5f))
* Add typing to proto.Message based class attributes ([a2a534c](https://github.com/googleapis/python-texttospeech/commit/a2a534ceccbad9114cb30599129eb168ab60cb5f))


### Bug Fixes

* Add dict typing for client_options ([a2a534c](https://github.com/googleapis/python-texttospeech/commit/a2a534ceccbad9114cb30599129eb168ab60cb5f))
* **deps:** Require google-api-core &gt;=1.34.0, >=2.11.0  ([07227a0](https://github.com/googleapis/python-texttospeech/commit/07227a06b555d590fc35055ec229f22c6d8dd139))
* Drop usage of pkg_resources ([07227a0](https://github.com/googleapis/python-texttospeech/commit/07227a06b555d590fc35055ec229f22c6d8dd139))
* Fix timeout default values ([07227a0](https://github.com/googleapis/python-texttospeech/commit/07227a06b555d590fc35055ec229f22c6d8dd139))


### Documentation

* **samples:** Snippetgen handling of repeated enum field ([a2a534c](https://github.com/googleapis/python-texttospeech/commit/a2a534ceccbad9114cb30599129eb168ab60cb5f))
* **samples:** Snippetgen should call await on the operation coroutine before calling result ([07227a0](https://github.com/googleapis/python-texttospeech/commit/07227a06b555d590fc35055ec229f22c6d8dd139))

## [2.12.3](https://github.com/googleapis/python-texttospeech/compare/v2.12.2...v2.12.3) (2022-10-07)


### Bug Fixes

* **deps:** Allow protobuf 3.19.5 ([#338](https://github.com/googleapis/python-texttospeech/issues/338)) ([aa92121](https://github.com/googleapis/python-texttospeech/commit/aa921215c05588b6555b7b41381b30f8b0079d54))

## [2.12.2](https://github.com/googleapis/python-texttospeech/compare/v2.12.1...v2.12.2) (2022-10-03)


### Bug Fixes

* **deps:** Require protobuf >= 3.20.2 ([#335](https://github.com/googleapis/python-texttospeech/issues/335)) ([50b394d](https://github.com/googleapis/python-texttospeech/commit/50b394dc5684b2df6cf544ba7bb5296d384ee0f8))

## [2.12.1](https://github.com/googleapis/python-texttospeech/compare/v2.12.0...v2.12.1) (2022-08-11)


### Bug Fixes

* **deps:** allow protobuf < 5.0.0 ([#316](https://github.com/googleapis/python-texttospeech/issues/316)) ([ec75f1e](https://github.com/googleapis/python-texttospeech/commit/ec75f1e5382bd527d17fb347a7ecc7cd35f57ab0))
* **deps:** require proto-plus >= 1.22.0 ([ec75f1e](https://github.com/googleapis/python-texttospeech/commit/ec75f1e5382bd527d17fb347a7ecc7cd35f57ab0))

## [2.12.0](https://github.com/googleapis/python-texttospeech/compare/v2.11.1...v2.12.0) (2022-07-16)


### Features

* add audience parameter ([f450551](https://github.com/googleapis/python-texttospeech/commit/f450551482e52c7f5564bf735b705e22255dc5d8))


### Bug Fixes

* **deps:** require google-api-core>=1.32.0,>=2.8.0 ([#304](https://github.com/googleapis/python-texttospeech/issues/304)) ([f450551](https://github.com/googleapis/python-texttospeech/commit/f450551482e52c7f5564bf735b705e22255dc5d8))
* require python 3.7+ ([#306](https://github.com/googleapis/python-texttospeech/issues/306)) ([192277b](https://github.com/googleapis/python-texttospeech/commit/192277b9b62338840b55fe8e24be582a19f390cd))

## [2.11.1](https://github.com/googleapis/python-texttospeech/compare/v2.11.0...v2.11.1) (2022-06-06)


### Bug Fixes

* **deps:** require protobuf <4.0.0dev ([#293](https://github.com/googleapis/python-texttospeech/issues/293)) ([e5ab0d6](https://github.com/googleapis/python-texttospeech/commit/e5ab0d6b69e008653b1500742ede3ead3a748f58))


### Documentation


owl-bot-staging/v1beta1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2023 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

ALL_PYTHON = [
    "3.7",
    "3.8",
    "3.9",
    "3.10",
    "3.11",
]

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")

BLACK_VERSION = "black==22.3.0"
BLACK_PATHS = ["docs", "google", "tests", "samples", "noxfile.py", "setup.py"]
DEFAULT_PYTHON_VERSION = "3.11"

nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
    "blacken",
    "lint",
    "lint_setup_py",
]

@nox.session(python=ALL_PYTHON)
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'pytest-asyncio', 'asyncmock; python_version < "3.8"')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/texttospeech_v1beta1/',
        '--cov=tests/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=ALL_PYTHON)
def mypy(session):
    """Run the type checker."""
    session.install(
        'mypy',
        'types-requests',
        'types-protobuf'
    )
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python=DEFAULT_PYTHON_VERSION)
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx==4.0.1", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def lint(session):
    """Run linters.

    Returns a failure if the linters find linting errors or sufficiently
    serious code quality issues.
    """
    session.install("flake8", BLACK_VERSION)
    session.run(
        "black",
        "--check",
        *BLACK_PATHS,
    )
    session.run("flake8", "google", "tests", "samples")


@nox.session(python=DEFAULT_PYTHON_VERSION)
def blacken(session):
    """Run black. Format code to uniform standard."""
    session.install(BLACK_VERSION)
    session.run(
        "black",
        *BLACK_PATHS,
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def lint_setup_py(session):
    """Verify that setup.py is valid (including RST check)."""
    session.install("docutils", "pygments")
    session.run("python", "setup.py", "check", "--restructuredtext", "--strict")
